### PR TITLE
feat: implements and handles verify phrase key stage to secure backup

### PR DESCRIPTION
**Pull request by domw30**
### What does this do?
- implements and handles verify key phrase stage to secure backup.

### Why are we making this change?
- to display the `VerifyKeyPhrase` component in the UI.

### How do I test this?
- run tests as usual / run locally and follow steps in demo below with a user that has not yet backed up key.

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

DEMO

https://github.com/zer0-os/zOS/assets/39112648/64c8787d-59cb-485a-9531-40c901ba99e0

### CODE REVIEW

File: src/components/secure-backup/index.test.tsx [hunk 1]
```diff
@@ -184,12 +185,12 @@ describe('SecureBackup', () => {
     });
 
     it('publishes onSave', function () {
-      const onSave = jest.fn();
-      const wrapper = subject({ backupStage: BackupStage.GenerateBackup, onSave });
+      const onVerifyKey = jest.fn();
```
Comment (domw30) [inline src/components/secure-backup/index.test.tsx:188]: Please note: I will be updating the names of props in a follow up PR.

File: src/components/secure-backup/index.tsx [hunk 2]
```diff
@@ -117,11 +118,14 @@ export class SecureBackup extends React.PureComponent<Properties> {
         );
 
       case BackupStage.GenerateBackup:
-        return <GenerateBackup recoveryKey={recoveryKey} errorMessage={errorMessage} onSave={onSave} />;
+        return <GenerateBackup recoveryKey={recoveryKey} errorMessage={errorMessage} onSave={onVerifyKey} />;
```
Comment (domw30) [inline src/components/secure-backup/index.tsx:121]: Please note: I will be updating the names of props in a follow up PR.

File: src/store/matrix/saga.ts [hunk 3]
```diff
@@ -27,6 +27,9 @@ export function* saga() {
   yield takeLatest(SagaActionTypes.SaveBackup, saveBackup);
   yield takeLatest(SagaActionTypes.RestoreBackup, restoreBackup);
   yield takeLatest(SagaActionTypes.ClearBackup, clearBackupState);
+  yield takeLatest(SagaActionTypes.OpenBackupDialog, openBackupDialog);
```
Comment (domw30) [inline src/store/matrix/saga.ts:30]: Moved these just to keep `For debugging` grouping

File: src/components/secure-backup/index.test.tsx [hunk 4]
```diff
@@ -226,6 +227,31 @@ describe('SecureBackup', () => {
...
+    it('renders VerifyKeyPhrase component when in VerifyKeyPhrase stage', function () {
+      const wrapper = subject({ backupStage: BackupStage.VerifyKeyPhrase });
+      expect(wrapper).toHaveElement(VerifyKeyPhrase);
+    });
+
+    it('publishes onGenerate', function () {
+      const onGenerate = jest.fn();
+      const wrapper = subject({ backupStage: BackupStage.VerifyKeyPhrase, onGenerate });
+
+      wrapper.find(VerifyKeyPhrase).simulate('back');
```
Comment (domw30) [inline src/components/secure-backup/index.test.tsx:240]: Please note: I will be updating the names of props in a follow up PR.

File: src/store/matrix/saga.ts [hunk 5]
```diff
@@ -72,17 +80,35 @@ export function* generateBackup() {
...
-export function* saveBackup() {
+export function* proceedToVerifyKey() {
+  const existingKey = yield select((state) => state.matrix.generatedRecoveryKey);
+
+  if (existingKey) {
+    yield put(setBackupStage(BackupStage.VerifyKeyPhrase));
+  } else {
+    yield put(setBackupStage(BackupStage.RestoreBackup));
+  }
+}
```
Comment (dalefukami) [inline src/store/matrix/saga.ts:91]: This seems like we've mixed two possible saga paths. They should just be separate.

**Review by dalefukami — APPROVED** [review]

File: src/store/matrix/saga.ts (hunk 5 above)
Comment (domw30) [reply src/store/matrix/saga.ts:91]: Yeah another picky one as they both 'verify' in a sense, but agree, could be improved

**Pull request merged**